### PR TITLE
Add integrations for ucaip

### PR DESCRIPTION
**Pull request by bovard**
Allow users to use the aiplatform.Datasets and aiplatform.CustomTrainingJob packages

### CODE REVIEW

File: patches/kaggle_gcp.py [hunk 1]
```diff
@@ -310,6 +310,24 @@ def init_natural_language():
...
+    from kaggle_gcp import get_integrations
+    if not get_integrations().has_cloudai():
+        return
+
+    from kaggle_secrets import GcpTarget
+    from kaggle_gcp import KaggleKernelCredentials
+    kaggle_kernel_credentials = KaggleKernelCredentials(target=GcpTarget.CLOUDAI)
+
+    # Patch the 2 ucaip clients: Datasets & CustomTrainingJob
+    monkeypatch_client(aiplatform.Datasets, kaggle_kernel_credentials)
```
Comment (rosbo) [inline patches/kaggle_gcp.py:327]: The other clients all have the `*Client` suffix: `AutoMlClient`, `LanguageServiceClient`, `PredictionServiceClient`.

Did you check that `Datasets` and `CustomTrainingJob` uses the same structure and ensured that `monkeypatch_client` works for these classes?

File: Dockerfile [hunk 2]
```diff
@@ -276,6 +276,10 @@ RUN pip install --upgrade cython && \
     pip install google-cloud-language==2.* && \
     pip install google-cloud-videointelligence==2.* && \
     pip install google-cloud-vision==2.* && \
+    # After launch install main
+    pip install git+https://github.com/googleapis/python-aiplatform.git@mb-release && \
```
Comment (rosbo) [inline Dockerfile:280]: Is there a prebuilt version of the python aiplatform on pip that we could use?

File: Dockerfile [hunk 3]
```diff
@@ -276,6 +276,10 @@ RUN pip install --upgrade cython && \
     pip install google-cloud-language==2.* && \
     pip install google-cloud-videointelligence==2.* && \
     pip install google-cloud-vision==2.* && \
+    # After launch install main
+    pip install git+https://github.com/googleapis/python-aiplatform.git@mb-release && \ 
+    pip install --upgrade pandas && \
```
Comment (rosbo) [inline Dockerfile:281]: This is probably breaking other packages. `pandas` is used by many packages, if you need a newer version, let me know and I will look into upgrading the DLVM base image to get a newer version.

I also see a comment below about removing this.

File: Dockerfile [hunk 4]
```diff
@@ -276,6 +276,10 @@ RUN pip install --upgrade cython && \
     pip install google-cloud-language==2.* && \
     pip install google-cloud-videointelligence==2.* && \
     pip install google-cloud-vision==2.* && \
+    # After launch install main
```
Comment (rosbo) [inline Dockerfile:279]: Not sure I understand what this comment mean.

**Review by rosbo — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (bovard) [reply Dockerfile:280]: there isn't unfortunately, the dev branches has the needed changes and won't be merged until launch

File: Dockerfile (hunk 3 above)
Comment (bovard) [reply Dockerfile:281]: I tried a couple different configurations and I *think* this one can be removed, I'll give it a shot

File: Dockerfile (hunk 4 above)
Comment (bovard) [reply Dockerfile:279]: I'll reword

**Review by rosbo — APPROVED** [review]

**Pull request merged**